Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift

```diff
@@ -120,21 +120,24 @@ public class PortfolioStore: ObservableObject {
   private let assetRatioService: BraveWalletAssetRatioService
   private let blockchainRegistry: BraveWalletBlockchainRegistry
   private let ipfsApi: IpfsAPI
+  private let assetManager: WalletUserAssetManagerType
 
   public init(
     keyringService: BraveWalletKeyringService,
     rpcService: BraveWalletJsonRpcService,
     walletService: BraveWalletBraveWalletService,
     assetRatioService: BraveWalletAssetRatioService,
     blockchainRegistry: BraveWalletBlockchainRegistry,
-    ipfsApi: IpfsAPI
+    ipfsApi: IpfsAPI,
+    userAssetManager: WalletUserAssetManagerType = WalletUserAssetManager()
```

Comment:
Should all these stores have a default value for `WalletUserAssetManagerType`? Currently if we don't pass through the existing store, we're initializing a new manager for each store it is used; I don't think it's a problem currently but when we later start caching balance, we could add a delegate to this `WalletUserAssetManager` to notify of balance updates or even asset visibility changes if needed.

Possible you have looked at balance caching while working on this and have another plan/solution?